scrypto-test/src/environment/builder.rs

```diff
@@ -200,6 +200,13 @@ where
             self.database.commit(&database_updates);
         }
 
+        // Getting the kernel boot to use for the kernel creation.
+        let kernel_boot = self.database.get_existing_substate::<KernelBoot>(
```

Comment:
I think you want `KernelBoot::load(&self.database)` which has the correct fallbacks when run on pre-bottlenose.

Reply:
Neat!